### ISSUE TITLE
Pattern variables are treated as oblivious

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#33499]
The following code should cause a warning when `null` is assigned to the non-nullable variable `s`, but no warning is produced.

``` c#
#nullable enable
class Test
{
    static void M(object o)
    {
        if (o is string s) { }
        s = null;
    }
}
```
